neuraxle/base.py

```diff
@@ -387,7 +419,21 @@ class MetaStepMixin:
     def __init__(self, wrapped: BaseStep = None):
         self.wrapped: BaseStep = wrapped
 
-    def set_hyperparams(self, hyperparams: HyperparameterSamples) -> 'BaseStep':
+    def setup(self, step_path: str, setup_arguments: dict = None) -> BaseStep:
+        name__ = MetaStepMixin.__name__
+
+        self.wrapped.setup(
+            step_path=os.path.join(step_path, '{0}({1})'.format(self.wrapped.name, name__)),
+            setup_arguments=setup_arguments
+        )
+
+        return self
+
+    @abstractmethod
+    def get_name(self):
+        return '{0}({1})'.format(self.wrapped.name,  MetaStepMixin.__name__)
+
+    def set_hyperparams(self, hyperparams: HyperparameterSamples) -> BaseStep:
         self.wrapped = self.wrapped.set_hyperparams(hyperparams)
```

Comment:
I know this isn't in this PR, but should we remove an "__" chain in the dict keys of the hyperparams before sending them to the substep?

Reply:
I believe so.

Reply:
won't fix here. please open an issue, I'm not sure what you mean here thanks